storm-core/src/clj/org/apache/storm/daemon/worker.clj

```diff
@@ -487,11 +486,12 @@
             )))))
 
 (defn register-callbacks [worker]
-  (log-message "Registering IConnectionCallbacks for " (:assignment-id worker) ":" (:port worker))
-  (msg-loader/register-callback (:transfer-local-fn worker)
-                                (:receiver worker)
-                                (:storm-conf worker)
-                                (worker-context worker)))
+  (let [transfer-local-fn (:transfer-local-fn worker)
+        ^IConnection socket (:receiver worker)]
+    (log-message "Registering IConnectionCallbacks for " (:assignment-id worker) ":" (:port worker))
+    (.registerRecv socket (DeserializingConnectionCallback. (:storm-conf worker)
```

Comment:
I am still not very convinced. I will port DeserializationCallBack to clojure, then we will again port it back to java. IMO it will be better to make it inline later, if needed. 
@revans2 Since you wrote DeserializationCallback, was there any particular reason for writing it outside worker.clj e.g. intending to use it in other places?